### PR TITLE
Vibhansa/proxy details

### PR DESCRIPTION
**Pull request by vibhansa-msft**
Correction proxy cli option details on when to skip that and when to use it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,8 +70,8 @@ For more information, see the [wiki](https://github.com/Azure/azure-storage-fuse
...
-     * [OPTIONAL] **--http-proxy=http://10.1.22.4:8080/** : Only used when https is turned off using --use-https=false, and if external network is only available through a proxy server, this parameter should specify the proxy server along with the port which is 8080 unless there are some deviations from normal port allocation numbers. This option is only available from version 1.3.7
+     * [OPTIONAL] **--https-proxy=http://10.1.22.4:8080/** : If external network is only available through a proxy server, this parameter should specify the proxy server along with the port which is 8080 unless there are some deviations from normal port allocation numbers. This option is only available from version 1.3.7. Use this cli option only when proxy needs a certificate for authentication to go through and certificate is not available at standard linux path. For regular proxy connection defining standard envrionment variables for proxy (http_proxy/https_proxy) are enough.
```
Comment (gapra-msft) [inline README.md:73]:
```suggestion
     * [OPTIONAL] **--https-proxy=http://10.1.22.4:8080/** : If external network is only available through a proxy server, this parameter should specify the proxy server along with the port which is 8080 unless there are some deviations from normal port allocation numbers. This option is only available from version 1.3.7. Use this cli option only when proxy needs a certificate for authentication to go through and certificate is not available at standard linux path. For regular proxy connection defining standard environment variables for proxy (http_proxy/https_proxy) are enough.
```

File: README.md [hunk 2]
```diff
@@ -70,8 +70,8 @@ For more information, see the [wiki](https://github.com/Azure/azure-storage-fuse
...
+     * [OPTIONAL] **--http-proxy=http://10.1.22.4:8080/** : Only used when https is turned off using --use-https=false, and if external network is only available through a proxy server, this parameter should specify the proxy server along with the port which is 8080 unless there are some deviations from normal port allocation numbers. This option is only available from version 1.3.7. Use this cli option only when proxy needs a certificate for authentication to go through and certificate is not available at standard linux path. For regular proxy connection defining standard envrionment variables for proxy (http_proxy/https_proxy) are enough.
```
Comment (gapra-msft) [inline README.md:74]:
```suggestion
     * [OPTIONAL] **--http-proxy=http://10.1.22.4:8080/** : Only used when https is turned off using --use-https=false, and if external network is only available through a proxy server, this parameter should specify the proxy server along with the port which is 8080 unless there are some deviations from normal port allocation numbers. This option is only available from version 1.3.7. Use this cli option only when proxy needs a certificate for authentication to go through and certificate is not available at standard linux path. For regular proxy connection defining standard environment variables for proxy (http_proxy/https_proxy) are enough.
```

File: blobfuse/blobfuse.cpp [hunk 3]
```diff
@@ -993,6 +993,12 @@ read_and_set_arguments(int argc, char *argv[], struct fuse_args *args)
     }
 
     config_options.tmpPath = tmpPathStr;
+    char real_path[PATH_MAX];
+    char *result = realpath(tmpPathStr.c_str(), real_path);
+    if (result != NULL) {
+        config_options.absoluateTmpPath = std::string(real_path);
```
Comment (gapra-msft) [inline blobfuse/blobfuse.cpp:999]: typo: absolute

**Review by gapra-msft — APPROVED** [review]

File: blobfuse/utilities.cpp [hunk 4]
```diff
@@ -183,6 +184,20 @@ int azs_getattr(const char *path, struct stat *stbuf)
             }
             else
             {
+                char real_path[PATH_MAX];
+                char *result = realpath(mntPathString.c_str(), real_path);
+                if (result != NULL) {
+                    // Path got resolved. lets check we are not moving away from temp path
+                    // If the file name was created using \..\..\..\abc.txt kind of pattern then mergin it with temp cache path
```
Comment (tasherif-msft) [inline blobfuse/utilities.cpp:191]:
```suggestion
                    // If the file name was created using \..\..\..\abc.txt kind of pattern then merging it with temp cache path
```

File: blobfuse/utilities.cpp [hunk 5]
```diff
@@ -183,6 +184,20 @@ int azs_getattr(const char *path, struct stat *stbuf)
...
+                    // may lead the path out of the temp cache folde rand blobfuse may end up deleting those file if open is hit
```
Comment (tasherif-msft) [inline blobfuse/utilities.cpp:192]:
```suggestion
                    // may lead the path out of the temp cache folder and blobfuse may end up deleting those files if open is hit
```

File: blobfuse/utilities.cpp [hunk 6]
```diff
@@ -183,6 +184,20 @@ int azs_getattr(const char *path, struct stat *stbuf)
...
+                    // To restrict that here we stop getattr and return enoent so that open is never called for such path
+                    // any absolute path going out of temp cache path needs to be invalidated
+                    std::string realPathString(real_path);
+                    if (std::string::npos == realPathString.find(config_options.absoluateTmpPath)) {
+                        AZS_DEBUGLOGV("Resolved absolute path is = %s for %s, and its out of scope\n", real_path, mntPathString.c_str());
```
Comment (tasherif-msft) [inline blobfuse/utilities.cpp:197]: We should also add this change in our PR description

**Review by tasherif-msft — APPROVED** [review]

File: blobfuse/utilities.cpp (hunk 6 above)
Comment (tasherif-msft) [reply blobfuse/utilities.cpp:197]: it looks like these changes are in a separate PR, you can ignore my change requests here and apply them in the other PR for this fix!

**Pull request merged**